### PR TITLE
Add Snabela 1.0, a logic-less template system

### PR DESCRIPTION
**Pull request by orbitz**

### CODE REVIEW

File: packages/snabela/snabela.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+]
+
+authors: [
+	"mmatalka@gmail.com"
+]
+
+homepage: "https://bitbucket.org/acslab/snabela"
+bug-reports: "https://bitbucket.org/acslab/snabela/issues"
+dev-repo: "git@bitbucket.org:acslab/snabela.git"
+
```
Comment (avsm) [inline packages/snabela/snabela.1.0/opam]: needs to be constrained to ocaml >=4.03.0 , or the `result` compatibility package needs to be added

```
# File "../../../src/snabela/snabela_lexer.ml", line 148, characters 4-6:
# Error: Unbound constructor Ok
# make[2]: *** [snabela_lexer.cmx] Error 2
# make[1]: *** [release_snabela] Error 2
# make: *** [all] Error 2
```

**Comment by orbitz** [comment]
I'm not sure if I broke it or not!  I updated the PR + code with `result` dep but I reused the same tag since I didn't consider `1.0` public until it gets merged.  But based on the errors I think something is up with the testing infra?

**Comment by avsm** [comment]
seems to fail on 4.02.3 with
```
snabela.1.0 =======================================#
# opam-version         1.2.2 (58ef3b8213100953848d362f7120a30356d7f77d)
# os                   linux
# command              make -j2
# path                 /home/opam/.opam/4.02.3/build/snabela.1.0
# compiler             4.02.3
# exit-code            2
# env-file             /home/opam/.opam/4.02.3/build/snabela.1.0/snabela-29-be7d52.env
# stdout-file          /home/opam/.opam/4.02.3/build/snabela.1.0/snabela-29-be7d52.out
# stderr-file          /home/opam/.opam/4.02.3/build/snabela.1.0/snabela-29-be7d52.err
### stdout ###
# [...]
# snabela.ml snabela_lexer.ml snabela.mli > /home/opam/.opam/4.02.3/build/snabela.1.0/build/release/snabela/.d
# ocamlfind ocamlopt -thread -I +camlp4 -package sedlex -bin-annot -w '@f@p@u@s@40' -strict-sequence -strict-formats -safe-string -noassert -package containers,ppx_deriving,ppx_deriving.eq,ppx_deriving.show,process,result,sedlex  -c ../../../src/snabela/snabela_lexer.ml -o snabela_lexer.cmx
# ocamlfind ocamlc -thread -I +camlp4 -package sedlex -bin-annot -w '@f@p@u@s@40' -strict-sequence -strict-formats -safe-string -noassert -package containers,ppx_deriving,ppx_deriving.eq,ppx_deriving.show,process,result,sedlex  -c ../../../src/snabela/snabela_lexer.ml -o snabela_lexer.cmo
# ocamlfind ocamlc -thread -I +camlp4 -package sedlex -bin-annot -w '@f@p@u@s@40' -strict-sequence -strict-formats -safe-string -noassert -package containers,ppx_deriving,ppx_deriving.eq,ppx_deriving.show,process,result,sedlex -c ../../../src/snabela/snabela.mli -o snabela.cmi
# ocamlfind ocamlopt -thread -I +camlp4 -package sedlex -bin-annot -w '@f@p@u@s@40' -strict-sequence -strict-formats -safe-string -noassert -package containers,ppx_deriving,ppx_deriving.eq,ppx_deriving.show,process,result,sedlex  -c ../../../src/snabela/snabela.ml -o snabela.cmx
# ../../../Ocamlrules.mk.in:113: recipe for target 'snabela.cmx' failed
# make[2]: Leaving directory '/home/opam/.opam/4.02.3/build/snabela.1.0/build/release/snabela'
# pds.mk:71: recipe for target 'release_snabela' failed
# make[1]: Leaving directory '/home/opam/.opam/4.02.3/build/snabela.1.0'
# Makefile:4: recipe for target 'all' failed
### stderr ###
# File "../../../src/snabela/snabela.ml", line 256, characters 14-26:
# Error: Unbound value Kv.Map.union
# make[2]: *** [snabela.cmx] Error 2
# make[1]: *** [release_snabela] Error 2
# make: *** [all] Error 2
```

is this too old a containers version or something similar?

**Comment by orbitz** [comment]
I decided it was easier to just not support 4.02.  I might go back later and support it.  Hopefully this latest push works.

**Comment by orbitz** [comment]
The test run might need some love to get going, if I'm understanding the output.

**Comment by orbitz** [comment]
@avsm when you get a chance, can you check why the latest push failed?

**Comment by avsm** [comment]
restarted jobs - looks like menhir's site was down

**Comment by orbitz** [comment]
I guess the menhir site is still down?

**Comment by avsm** [comment]
enough has passed that this is good to go :-)

**Pull request merged**